src/services/localizations/readFiles.ts

```diff
@@ -0,0 +1,53 @@
+import { readdirSync, readFileSync } from 'fs';
+import * as osPath from 'path';
+import { PartialLocalization } from '../../repositories/localizations';
+import { assertValidFileContent } from './assertValidFileContent';
+import { assertValidLanguageCode } from './assertValidLanguageCode';
+
+/**
+ * Read all localization files from a directory and transform them into an array of localizations
+ */
+export function readFiles(path: string) {
+  let fileNames: string[];
+  try {
+    fileNames = readdirSync(path);
+  } catch (error) {
+    throw new Error(`Was not able to list localization files in directory '${path}': ${error}`);
+  }
+
+  const localizationMap: Record<string, PartialLocalization> = {};
+  for (const fileName of fileNames) {
+    const parsedFileName = osPath.parse(fileName);
+
+    // Only process the .json files!
+    if (parsedFileName.ext !== '.json') {
+      continue;
+    }
+
+    const languageCode = parsedFileName.name.toUpperCase();
+    assertValidLanguageCode(languageCode);
+
+    const localizationJson = readLocalizationFileSync(path, fileName);
+    assertValidFileContent(fileName, localizationJson);
+
+    for (const [key, text] of Object.entries(localizationJson)) {
```

Comment:
This is a super dumb ass comment... but you've put in the work to make a beautiful predicate for the type systems inference.

IF you wrap this for loop in an IF statement you retain valid typing instead of any. (You really don't have to do this, but you've already put in the effort, this is like 10000000% pedantic)

```
    if (assertValidFileContent(fileName, localizationJson)) {
      for (const [key, text] of Object.entries(localizationJson)) {
        if (!(key in localizationMap)) {
          localizationMap[key] = {key, text: {}};
        }

        localizationMap[key].text[languageCode] = text;
      }
    }
  }
  ```

Reply:
Fair fair fair. As I'm allergic to indentation, I have fixed it by making the type guard a `asserts` type guard (as mentioned above).